ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py

```diff
@@ -1152,3 +1153,11 @@ def get_script(datasource, context):
         log.error('Invalid tales expression in custom command script: %s' %
                   str(datasource.script))
     return script
+
+
+def get_valid_dsconf(dsconfs):
+    '''return a valid eventClass'''
+    for dsconf in dsconfs:
+        if dsconf.eventClass != '':
+            return dsconf.eventClass
+    return None
```

Comment:
Why do you need `return None` at the end?
One more thing, you have the line `eventClass = dsconf.eventClass or get_eventClass`. It looks like we'll have `eventClass = None` if `get_valid_dsconf` function returns nothing.

Reply:
Agree, that should be fixed